src/main/java/org/gridsuite/geodata/server/GeoDataController.java

```diff
@@ -50,17 +51,23 @@ private static Set<Country> toCountrySet(@RequestParam(required = false) List<St
     }
 
     @GetMapping(value = "/substations", produces = MediaType.APPLICATION_JSON_VALUE)
-    @Operation(summary = "Get substations geographical data")
+    @Operation(summary = "Get geographical data for substations with the given ids")
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Substations geographical data")})
     public ResponseEntity<List<SubstationGeoData>> getSubstations(@Parameter(description = "Network UUID")@RequestParam UUID networkUuid,
+                                                                  @RequestParam(required = false) List<String> countries,
                                                                   @Parameter(description = "Variant Id") @RequestParam(name = "variantId", required = false) String variantId,
-                                                                  @Parameter(description = "Countries")@RequestParam(required = false) List<String> countries) {
+                                                                  @Parameter(description = "Substations ids") @RequestParam(name = "substationId", required = false) List<String> substationsIds) {
         Set<Country> countrySet = toCountrySet(countries);
-        Network network = networkStoreService.getNetwork(networkUuid);
+        Network network = networkStoreService.getNetwork(networkUuid, substationsIds != null ? PreloadingStrategy.NONE : PreloadingStrategy.COLLECTION);
         if (variantId != null) {
             network.getVariantManager().setWorkingVariant(variantId);
         }
-        List<SubstationGeoData> substations = geoDataService.getSubstations(network, countrySet);
+        List<SubstationGeoData> substations;
+        if (substationsIds != null) {
+            substations = geoDataService.getSubstations(network, substationsIds);
```

Comment:
May warn if countrySet is not empty, no ?

Reply:
It is not required now. Filter by ids is already very restrictive.
I added the warning in case substation ids AND countries are given by parameters.
Same for lines